DataDog/datadog-agent
# Write sysv-init scripts for Agent 6, on Debian

### What does this PR do?

Write sysv-init scripts for Agent 6, on Debian

### Motivation

Support Debian installation not using systemd

Notes:
- trace and process agents are started when the infra agent has started in the datadog-agent script.
- the agent is not restarted on failure unlike with the systemd service

When present on an upstart or systemd system, these scripts are never used and doesn't interfere at all with the native upstart or systemd services.
- upstart: There is a special function to detect upstart presence and to do nothing in this case (using https://wiki.ubuntu.com/UpstartCompatibleInitScripts)
- systemd: Use init.d script only if it doesn't find a native service. Since we ship a systemd service, it's ok

### Tests
- Tested on debian 7 with sysvinit, systemd and upstart, both by adding the scripts manually and using the package.
- Tested installing the package on a clean VM and upgrading from a version that doesn't support sysvinit 

### QA
- Install the package on a clean VM using sysvinit/systemd/upstart
- Upgrade from a package that doesn't support sysvinit on a VM using sysvinit/systemd/upstart
- Start, Stop, Restart, Enable, Disable the services
- Check that the agent starts at boot
- Check that this scripts don't interfere with the existing upstart/systemd services
